### PR TITLE
Sync ignored v4 presign headers

### PR DESCRIPTION
**Pull request by superseb**
https://github.com/minio/minio-go/issues/1629

### CODE REVIEW

File: pkg/signer/request-signature-v4.go [hunk 1]
```diff
@@ -46,21 +46,26 @@ const (
...
+// * Accept-Encoding
+// Some S3 servers like Hitachi Content Platform do not honor this header for signature
+// calculation.
 //
 var v4IgnoredHeaders = map[string]bool{
-	"Authorization": true,
-	"User-Agent":    true,
+	"Accept-Encoding":      true,
+	"Authorization":        true,
+	"Content-MD5":          true,
```
Comment (harshavardhana) [inline pkg/signer/request-signature-v4.go:64]: This is not an ignored header..

File: pkg/signer/request-signature-v4.go [hunk 2]
```diff
@@ -46,21 +46,26 @@ const (
...
+	"User-Agent":           true,
+	"X-Amz-Content-SHA256": true,
```
Comment (harshavardhana) [inline pkg/signer/request-signature-v4.go:66]: This is not an ignored header.

File: pkg/signer/request-signature-v4.go [hunk 3]
```diff
@@ -46,21 +46,26 @@ const (
...
+	"X-Amz-Date":           true,
```
Comment (harshavardhana) [inline pkg/signer/request-signature-v4.go:67]: This is not an ignored header.

File: pkg/signer/request-signature-v4.go [hunk 4]
```diff
@@ -46,21 +46,26 @@ const (
...
+	"X-Amz-Security-Token": true,
```
Comment (harshavardhana) [inline pkg/signer/request-signature-v4.go:68]: This is not an ignored header.

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by harshavardhana** [comment]
All you need to add is `accept-encoding` to the list.

**Comment by superseb** [comment]
@harshavardhana Thanks! I have synced the list from https://github.com/minio/minio-java/blob/master/api/src/main/java/io/minio/Signer.java#L58, let me know if you still just want to add `accept-encoding`.

**Comment by harshavardhana** [comment]
> @harshavardhana Thanks! I have synced the list from https://github.com/minio/minio-java/blob/master/api/src/main/java/io/minio/Signer.java#L58, let me know if you still just want to add `accept-encoding`.

ignoring those headers is a bug in minio-java @superseb please fix the PR here the way I asked.

**Comment by harshavardhana** [comment]
Please resubmit once the relevant changes are done closing.

**Pull request closed without merging**